### PR TITLE
Add a check for the visibility of repeated entity's parents

### LINKED ISSUE #1886: Applicant changing answer connected to visibility condition issue
Visibility condition bug reported by Brenda where changing an answer connected with a visibility condition and trying to go back results in errors.

Steps to reproduce:
- Set a visibility condition for a screen 
- From the applicant side, go through the form to get to the conditional screen 
- Enter data on conditional screen 
- Go back and change the original condition
- Conditional screen continues to show and has the data in it  
- If you delete the data and try to proceed it does not work if the form requires the data.  

Solution: CiviForm should delete the data entered in a conditional screen if the applicant changes the visibility condition field; sllow applicant to continue in the form.

Screenshot attached:
Visibility condition issue

### PR DESCRIPTION
**Pull request by kmcnellis**
### Description
When assembling a program, pass a repeated entity's block's visibility to the entity.  Then when testing if a repeated entity block is visible, test all its parent's conditions as well.

### Checklist
- [x] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)
Fixes #1886

### CODE REVIEW

File: server/app/services/applicant/ReadOnlyApplicantProgramServiceImpl.java [hunk 1]
```diff
@@ -267,18 +271,29 @@ private ImmutableList<Block> getBlocks(
   }
 
   private boolean showBlock(Block block) {
+    if (block.getRepeatedEntity().isPresent()) {
```
Comment (bion) [inline server/app/services/applicant/ReadOnlyApplicantProgramServiceImpl.java:274]: suggestion (non-blocking): consider adding a commenting mentioning that this is examining of the visibility of the entity's parents

File: server/app/services/applicant/RepeatedEntity.java [hunk 2]
```diff
@@ -92,12 +113,23 @@ public int depth() {
     return 1 + parent().map(RepeatedEntity::depth).orElse(0);
   }
 
+  /** Gets visibility for this and all parents of the repeated entity. */
+  public ImmutableList<PredicateDefinition> nestedVisibility() {
+    return Stream.concat(
+            parent()
+                .map(p -> p.nestedVisibility())
```
Comment (bion) [inline server/app/services/applicant/RepeatedEntity.java:120]: suggestion (non-blocking): the type would be more clear here with method reference i.e. `.map(PredicateDefinition::nestedVisibility)`

**Review by bion — APPROVED** [review]

**Review by clouser2 — APPROVED** [review]

File: server/app/services/applicant/RepeatedEntity.java [hunk 3]
```diff
@@ -92,12 +113,23 @@ public int depth() {
...
+  }
+
   /**
    * Contextualize the text with repeated entity names.
    *
    * <p>Replaces "\$this" with this repeated entity's name. "\$this.parent" and
-   * "\$this.parent.parent" (ad infinitum) are replaced with the names of the ancestors of this
-   * repeated entity.
+   * "\$this.parent.parent" (ad infinitum) are replaced with the names of theUserRepository.java:89
+   * ancestors of this repeated entity.
```
[the remark this answers is not in the document]
Comment (kmcnellis) [reply server/app/services/applicant/RepeatedEntity.java:132]: Yep!

File: server/app/services/applicant/RepeatedEntity.java (hunk 2 above)
Comment (kmcnellis) [reply server/app/services/applicant/RepeatedEntity.java]: done

**Pull request merged**